### PR TITLE
Enhancements for autoencoder

### LINKED ISSUE #176: Autoencoder - Early Stopping Implementation
Add early stopping functionality to optimize model training. This feature will help prevent overfitting and reduce unnecessary training time by monitoring key metrics and stopping training when no improvement is observed.

### PR DESCRIPTION
**Pull request by ngorrinm**
This PR introduces several improvements to the `AutoEncoder` class. The key enhancements include:

- **Early Stopping configuration**: The class now allows to enable or disable early stopping and set the patience parameter. This helps prevent overfitting by stopping training when validation loss stops improving.
- **Configure activation function**: Added the option to configure the activation function for both the encoder and decoder layers. Previously, the activation function was set to None by default.
- **Z-Score normalization**: Added an option to normalize input data using Z-score normalization (`mean` and `std`), in addition to the existing Min-Max scaling.
- **Configurable optimizers**: The model now supports multiple optimizers (`adam`, `sgd`, `rmsprop`, etc.), allowing users to experiment with different training strategies.

Resolves #176 and #185

### CODE REVIEW

File: mango_time_series/mango_time_series/models/autoencoder.py [hunk 1]
```diff
@@ -447,16 +595,17 @@ def reconstruct(self):
...
+            x_val_converted = x_val_converted * (scale_max - scale_min) + scale_min
+            x_test_converted = x_test_converted * (scale_max - scale_min) + scale_min
+        elif self.normalization_method == "zscore":
+            x_train_converted = x_train_converted * scale_std + scale_mean
+            x_val_converted = x_val_converted * scale_std + scale_mean
+            x_test_converted = x_test_converted * scale_std + scale_mean
 
         x_converted = np.concatenate(
-            (x_train_converted, x_val_converted, x_test_converted), axis=1
+            (x_train_converted.T, x_val_converted.T, x_test_converted.T), axis=1
```
Comment (gvalleg) [inline mango_time_series/mango_time_series/models/autoencoder.py:608]: Why this change?

**Review by gvalleg — APPROVED** [review]
All correct!
